docs/developer/developer-getting-started.md

```diff
@@ -0,0 +1,283 @@
+# Overview
+
+The intent of this document is to be a high level quick start guide to get up and running quickly. For a more in depth review on configuration please see [CONTRIBUTING.MD](../../CONTRIBUTING.md#running-the-ratify-cli).
+
+# Getting started
+
+High-level steps of this document:
+
+1. Start devcontainer
+1. Set aliases and environment variables
+1. Build images
+1. Create Kind cluster
+1. Install Gatekeeper (from repo via Helm)
+1. Install Ratify (from locally built image)
+1. Install a verifier
+1. Apply constraints and templates
+1. Validate
+
+## Devcontainer
+
+Start the devcontainer using the command pallet or the green button in the lower left corner of VSCode.
+
+## Create aliases
+
+```bash
+alias k="kubectl"
+```
+
+## Export variables
+
+```bash
+export RATIFY_NAMESPACE=ratify-service
+export KUBERNETES_VERSION=1.25.4
+export IMAGE_PULL_POLICY=IfNotPresent
+export RATIFY_LOG_LEVEL=INFO
+```
+
+## Build images
+
+### Ratify
+
+```bash
+docker build \
+    --progress=plain \
+    --no-cache \
+    -f ./httpserver/Dockerfile \
+    -t localbuild:test .
+```
+
+### CRDs
+
+```bash
+docker build \
+    --progress=plain \
+    --no-cache \
+    --build-arg KUBE_VERSION=${KUBERNETES_VERSION} \
+    --build-arg TARGETOS="linux" \
+    --build-arg TARGETARCH="amd64" \
+    -f crd.Dockerfile \
+    -t localbuildcrd:test ./charts/ratify/crds
+```
+
+## Kind
+
+### Create cluster
+
+```bash
+kind create cluster
+```
+
+### Delete cluster
+
+```bash
+kind delete cluster
+```
+
+### Load images into kind
+
+```bash
+kind load docker-image --name kind localbuild:test
+kind load docker-image --name kind localbuildcrd:test
+```
+
+## Gatekeeper
+
+### Add repo
+
+```bash
+helm repo add gatekeeper https://open-policy-agent.github.io/gatekeeper/charts
+```
+
+### Install
+
+```bash
+helm install gatekeeper/gatekeeper  \
+    --name-template=gatekeeper \
+    --namespace gatekeeper-system --create-namespace \
+    --set enableExternalData=true \
+    --set validatingWebhookTimeoutSeconds=7 \
+    --debug
+```
+
+## Ratify
+
+### Install
+
+Install Ratify using TLS and a self signed cert.
+
+Notes:
+
+- See [other ways](../../CONTRIBUTING.md#deploy-from-local-helm-chart)  to install and TLS/mTLS options.
+- If changes are made to a plugin, they will have to be re built using `make build-plugins` and copy the output to `./ratify/plugins/`.
+
+1. Supply a certificate to use with Ratify (httpserver) or use the following script to create a self-signed certificate.
+
+    ```bash
+    ./scripts/generate-tls-certs.sh ${RATIFY_NAMESPACE}
+    ```
+
+1. Install ratify using a certificate
+
+```bash
+helm install ratify ./charts/ratify \
+    --namespace ${RATIFY_NAMESPACE} --create-namespace \
+    --atomic \
+    --set provider.auth="tls" \
```

Comment:
We just merged some changes to support GK 3.11. part of that we removed this provider.auth field since the TLS type can be determined dynamically from the existence of the clientCA secret from GK.